### DISCUSSION TITLE
How to stop sending sdk_name label?

### DISCUSSION THREAD

**Discussion by mk48** [grafana/faro-web-sdk#271] [Q&A] [answered]
The sdk_name label is larger than my whole log lines :). I don't need any of the text in the sdk_name. How to stop sending that log information?
I am using faro 1.2.0

![image](https://github.com/grafana/faro-web-sdk/assets/3082469/456423d4-23fc-4d2a-9a20-f7d428124b62)

![image](https://github.com/grafana/faro-web-sdk/assets/3082469/3104c49e-aca4-4600-86df-2dc3768ae4fe)

**Comment by codecapitano** [comment, marked as answer]
Hi @mk48 

This is a known "problem" (unnecessary data which adds to each beacon payload).
Midterm we plan to trim down Faro page loads even more.

Currently there's no setting to disable this, so you need a little workaround, see example below. 


**Remove the SDK meta data** 
Faro packages different meta data in so called Meta object, like MetaBrowser, MetaApp and so on.
The object containing SDK meta data is the MetaSDK object.

You can remove the respective meta by using the `beforeSend()` hook.

```
initializeFaro({
    url: ...,
    ...
    beforeSend(item: TransportItem) {
      delete item.meta.sdk;
      return item;
    },
});
``` 


PS:
Nevertheless I think making the sdk meta data optional (and maybe disable it by default) is a low hanging fruit we can tackle earlier.
@cedricziel Wdyt?

**Reply by mk48** [reply]
@codecapitano Thanks, It's working after using the beforeSend() method.

**Reply by codecapitano** [reply]
Tracked here: https://github.com/grafana/faro-web-sdk/issues/305
